File: tests/glm/debug-diagnostics.ts
import { createDataFrame } from "../../src/dataframe/ts/dataframe/index.ts";
import { glm } from "../../src/dataframe/ts/wasm/glm-functions.ts";

const data = createDataFrame([
  { y: 5, x: 1 },
  { y: 7, x: 2 },
  { y: 9, x: 3 },
  { y: 11, x: 4 },
  { y: 13, x: 5 },
]);

const model = glm({
  formula: "y ~ x",
  family: "gaussian",
  link: "identity",
  data,
});

const result = model.getRawResult();
console.log("\n=== Debug Info ===");
console.log("Dispersion parameter:", result.dispersion_parameter);
console.log("Residual deviance:", result.deviance);
console.log("DF residual:", result.df_residual);
console.log("Residual std error:", result.residual_standard_error);
console.log("\nDeviance residuals:", result.deviance_residuals);
console.log("Leverage values:", result.leverage);

const residuals = model.residuals({ type: "deviance" });
const rstandard = model.rstandard({ type: "deviance" });
const rstudent = model.rstudent();

console.log("\n=== Results ===");
console.log("Residuals:", residuals);
console.log("Rstandard:", rstandard);
console.log("Rstudent:", rstudent);

File: packages/testing/glm/debug-diagnostics.R
data <- data.frame(
  y = c(5, 7, 9, 11, 13),
  x = c(1, 2, 3, 4, 5)
)

model <- glm(y ~ x, family = gaussian(link = "identity"), data = data)
summary_model <- summary(model)

cat("\n=== Debug Info (R) ===\n")
cat("Dispersion parameter:", summary_model$dispersion, "\n")
cat("Residual deviance:", deviance(model), "\n")
cat("DF residual:", df.residual(model), "\n")

cat("\nDeviance residuals:", residuals(model, type = "deviance"), "\n")
cat("Leverage values:", hatvalues(model), "\n")

cat("\n=== Results (R) ===\n")
cat("Residuals:", residuals(model, type = "deviance"), "\n")
cat("Rstandard:", rstandard(model, type = "deviance"), "\n")
cat("Rstudent:", rstudent(model), "\n")
